File: 09-08-heap-y-cola-prio/Main.java
import adts.priority_queue.*;
import java.util.Scanner;

public class Main {
   public static void main(String[] args) {
      System.err.println("Hola esto es un error");

      System.out.println("Cuantos elementos queres agregar?");

      Scanner sc = new Scanner(System.in);
      int N = sc.nextInt();

      PriorityQueue<String, Integer> pq = new MinPriorityQueue<>(N);

      for (int i = 0; i < N; i++) {
         String elem = sc.next();
         int prio = sc.nextInt();
         pq.push(elem, prio);
         System.out.println("Elemento mas chico es: " + pq.top());
         System.out.println("El heap contiene " + pq.size() + " elementos");
      }

      while (!pq.isEmpty()) {
         System.out.println("Elemento mas chico es " + pq.pop());
      }

      sc.close();
   }
}

File: 09-08-heap-y-cola-prio/main.cpp
#include "adts/priority_queue/min_priority_queue.cpp"
#include <iostream>
#include <ostream>
#include <string>

int main() {

  std::cerr << "Hola esto es un error" << std::endl;
  std::cout << "Cuantos elementos queres agregar?" << std::endl;
  int N;
  std::cin >> N;
  priorityQueue<std::string, int> *pq =
      new minPriorityQueue<std::string, int>(N);

  for (int i = 0; i < N; i++) {
    std::string elem;
    int prio;
    std::cin >> elem >> prio;
    pq->push(elem, prio);
    std::cout << "Elemento mas chico es: " << pq->top() << std::endl;
    std::cout << "El heap contiene " << pq->size() << " elementos" << std::endl;
  }

  while (!pq->isEmpty()) {
    std::cout << "Elemento mas chico es " << pq->pop() << std::endl;
  }
}
